+rem remote deployment whitelist feature

ktoso opened the pull request:
Initial working version, will do a bit of cleanup soon.

ktoso committed:
+rem remote deployment whitelist feature (initial commit)

+rem remote deployment whitelist feature (initial commit)

ktoso commented:
Resolves #21663

ktoso commented:
Oh and needs docs.

patriknw commented on akka-remote/src/test/scala/akka/remote/RemoteDeploymentWhitelistSpec.scala:
```diff
@@ -0,0 +1,193 @@
+
+  private def byteStringOfSize(size: Int) = ByteString.fromArray(Array.fill(size)(42: Byte))
+
+  val maxPayloadBytes = system.settings.config.getBytes("akka.remote.test.maximum-payload-bytes").toInt
```
cleanup the test, copy pasta from something else now, I think

ktoso commented:
Cleaned up

ktoso committed:
+doc,rem added docs on remoting whitelist

ktoso commented:
Added docs (and addressed missing docs in foldAsync/scanAsync - let me smuggle those through in this PR...)

![remoting_ _akka_documentation](https://cloud.githubusercontent.com/assets/120979/19517929/60c443c4-9604-11e6-862d-0db14b3ded74.png)

patriknw commented:
wat, do you have a quota on number of PRs?

ktoso commented:
Heh ok, splitting it out 😉

ktoso commented:
Split out to: https://github.com/akka/akka/pull/21701

ktoso commented:
Can I merge this?

patriknw commented on akka-docs/rst/java/remoting.rst:
```diff
@@ -168,6 +168,23 @@ you can advise the system to create a child on that remote node like so:
+Remote deployment whitelist
+---------------------------
+
+As remote deployment can potentially be abused aby both users and even attackers a whitelist feature
```
s/aby/by/

patriknw reviewed (approved):
LGTM

ktoso committed:
Update remoting.rst

ktoso committed:
Update remoting.rst

drewhk commented:
LGTM
